api/core/utils.py

```diff
@@ -57,78 +82,86 @@ def create_connection(
+def _get_parameter(connection: ConnectionModel) -> Dict[str, Parameter]:
+    parameters: Dict[str, Parameter] = {
+        "incoming": Parameter(),
+        "outgoing": Parameter(),
     }
-
-
-def _get_parameter(connection: ConnectionModel) -> Dict:
-    parameters: Dict = _get_standard_parameter_values()
     if connection.parameters:
         for parameter in connection.parameters:
-            parameters[parameter.parameter_name] = parameter.value
+            if parameter.direction == "incoming":
+                setattr(
+                    parameters["incoming"], parameter.parameter_name, parameter.value
+                )
+            else:
+                setattr(
+                    parameters["outgoing"], parameter.parameter_name, parameter.value
+                )
     return parameters
 
 
-def get_all_connections() -> List[Dict]:
+def get_all_connections() -> ConnectionList:
     connections: List[ConnectionModel] = wemulate_utils.get_connection_list()
-    connection_information: List[Dict] = []
+    connection_information: ConnectionList = ConnectionList(connections=[])
     for connection in connections:
-        connection_information.append(
-            {
+        complete_connection = ConnectionComplete(
+            **{
                 "connection_name": connection.connection_name,
                 "connection_id": connection.connection_id,
                 "first_logical_interface_id": connection.first_logical_interface_id,
                 "second_logical_interface_id": connection.second_logical_interface_id,
                 **_get_parameter(connection),
             }
         )
+        connection_information.connections.append(complete_connection)
     return connection_information
 
 
 def _reset_connection(connection_name: str) -> None:
     wemulate_utils.reset_connection(connection_name)
 
 
-def _detect_applied_parameters(parameter: Dict) -> Dict:
+def _detect_applied_parameters(parameters: Dict[str, int]) -> Dict[str, int]:
     applied_parameter: Dict = {}
-    if parameter["jitter"] != STANDARD_JITTER:
-        applied_parameter["jitter"] = parameter["jitter"]
-    if parameter["bandwidth"] != STANDARD_BANDWIDTH:
-        applied_parameter["bandwidth"] = parameter["bandwidth"]
-    if parameter["delay"] != STANDARD_DELAY:
-        applied_parameter["delay"] = parameter["delay"]
-    if parameter["packet_loss"] != STANDARD_PACKET_LOSS:
-        applied_parameter["packet_loss"] = parameter["packet_loss"]
+    if parameters[JITTER] != DEFAULT_PARAMETERS[JITTER]:
+        applied_parameter[JITTER] = parameters[JITTER]
+    if parameters[BANDWIDTH] != DEFAULT_PARAMETERS[BANDWIDTH]:
+        applied_parameter[BANDWIDTH] = parameters[BANDWIDTH]
+    if parameters[DELAY] != DEFAULT_PARAMETERS[DELAY]:
+        applied_parameter[DELAY] = parameters[DELAY]
+    if parameters[PACKET_LOSS] != DEFAULT_PARAMETERS[PACKET_LOSS]:
+        applied_parameter[PACKET_LOSS] = parameters[PACKET_LOSS]
```

Comment:
I would do something like this (pseudo code)
```
for param, default_value in DEFAULT_PARAMETERS:
    if parameters[param] != default_value:
        applied_parameter[param] = parameters[param]
```
But not sure if this is what you want

Reply:
better use `parameters.get(param)` to prevent KeyError